src/main/groovy/com/avioconsulting/mule/linter/model/MuleComponent.groovy

```diff
@@ -0,0 +1,45 @@
+package com.avioconsulting.mule.linter.model
+
+class MuleComponent {
+
+    private String name
+    private Map<String, String> attributes = [:]
+    private final List<MuleComponent> children
+
+    MuleComponent(Map<String, String> attributes) {
+        this(attributes, null)
+    }
+
+    MuleComponent(Map<String, String> attributes, List<MuleComponent> children) {
+        this.attributes = attributes
+        this.children = children
+    }
+
+    Boolean hasAttributeValue(String name) {
+        return attributes.get(name)?.length() > 0
+    }
+
+    String getName() {
+        if (name == null || name.length() < 1) {
+            name = attributes.get('name')
+        }
+        return name
+    }
```

Comment:
We should probably differentiate between the name of the component and an attribute called name.  We should have the XML element name and namespace as part of the MuleComponent for sure, then any attribute called name would be different.

Reply:
Spoke with Nisarg, he's going to make those changes in his branch.  His code relies on MuleComponent more...